packages/jaeger-ui/src/components/TracePage/TraceGraph/OpNode.js

```diff
@@ -0,0 +1,129 @@
+// @flow
+
+// Copyright (c) 2017 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as React from 'react';
+import { Popover } from 'antd';
+import colorGenerator from '../../../utils/color-generator';
+
+import type { PVertex } from '../../../model/trace-dag/types';
+
+import './OpNode.css';
+
+type Props = {
+  count: number,
+  errors: number,
+  time: number,
+  percent: number,
+  selfTime: number,
+  percentSelfTime: number,
+  operation: string,
+  service: string,
+  mode: string,
+};
+
+export const MODE_SERVICE = 'service';
+export const MODE_TIME = 'time';
+export const MODE_SELFTIME = 'selftime';
+
+export function round2(percent: number) {
```

Comment:
[`Number#toFixed()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed) should do the same thing, since thjs is for formatting the percents.

Reply:
As I understand toFixed(2) will always add 2 decimal places even when not needed.
So 40 % will render 40.00 %. Do you really want this?

Reply:
You're right. I'd say the only downside of `40 %` is that it will look smaller than most of the values, e.g. `12.34 %`, at first glance. I'd suggest changing to `value.toFixed(1)` but seems fine either way as it was a nit to start with 😄

Reply:
Will keep the current implementation. Self time in leafs are always `100 %` so this would look like `100.0 %` or `100.00 %`.